packages/algorand-js/src/runtime/account.ts

```diff
@@ -57,15 +55,8 @@ export class StoreAccount implements StoreAccountI {
    */
   getLocalState (appId: number, key: Uint8Array): StackElem | undefined {
     const localState = this.appsLocalState;
-    const data = localState.get(appId)?.["key-value"]; // can be undefined (eg. app opted in)
-    if (data) {
-      const keyValue = data.find(schema => compareArray(schema.key, key));
-      const value = keyValue?.value;
-      if (value) {
-        return value.type === SSC_BYTES ? value.bytes : BigInt(value.uint);
-      }
-    }
-    return undefined;
+    const data = localState.get(appId)?.[keyValue]; // can be undefined (eg. app opted in)
+    return data?.get(key.toString());
```

Comment:
what happens if localstate is undefined ? does TEAL throws an error or returns 0 ?

Reply:
It pushes 0 to stack.

Reply:
okay, please update it to as TEAL behaves

Reply:
That part is handled in the opcode itself. If we receive `undefined` in `AppLocalGet` opcode, it pushes `0n` to stack.

Reply:
when we are testing TEAL, i would prefer it to behave same as TEAL behaves, 
because for some TEAL codes, if user wants to check what is there in localstate or how teal interprets it ? then it is  better.
what do you think ? what are the benefits of not doing it ?

Reply:
can we check for particular values from goal ?

Reply:
Agreed, no issues. It's just that we only interact with TEAL during a transaction execution, but we interact with runtime throughout the tests, that's why i was returning `undefined` here, and keeping teal relevant logic in opcodes. 
Will update it as per TEAL behaviour, and resolve similar comments.

Reply:
please add doc string for the same.

Reply:
> can we check for key-values(map) from goal ?

Yes, we can view account's local and global storage using `goal`. I think using `goal app read --local --app-id ..`

Reply:
Ok thanks, I couldn't find searching from key. For ex : finding value from map with a specific key. I think this must be the only way.

Reply:
> Ok thanks, I couldn't find searching from key. For ex : finding value from map with a specific key. I think this must be the only way.

Yes, i don't think we can do that. We can view complete global and local state only.

Reply:
Upon checking furthur, i think this can cause discrepency with `GetEx` opcodes, as we push did exist also with them. @amityadav0 can you please check: [https://github.com/scale-it/algorand-builder/blob/master/packages/algorand-js/src/interpreter/opcode-list.ts#L1581](https://github.com/scale-it/algorand-builder/blob/master/packages/algorand-js/src/interpreter/opcode-list.ts#L1581).

If we return `0n` here, then we won't know whether the value exists or not. One possible solution could be to return flag as well, but i am not sure if we should do that.

Reply:
maybe we can keep accounts functions as it is and update runtime functions to return 0, but if it causes discrepancy I agree with you, let's keep it as it is.

Reply:
Please add todo that we need to pass address to global state function in updated single storage.

Reply:
ok.